api/api.go

```diff
@@ -1,7 +1,45 @@
 package api
 
-import "go.sia.tech/core/types"
+import (
+	"go.sia.tech/core/net/gateway"
+
+	"go.sia.tech/core/types"
+	"go.sia.tech/siad/v2/wallet"
+)
 
 type WalletBalance struct {
 	Siacoins types.Currency `json:"siacoins"`
 }
+
+type WalletAddress struct {
+	Address types.Address `json:"address"`
+}
+
+type WalletAddresses struct {
+	Addresses []types.Address `json:"addresses"`
+}
+
+type WalletSignData struct {
+	ToSign      []types.ElementID `json:"toSign"`
+	Transaction types.Transaction `json:"transaction"`
+}
+
+type WalletTransaction struct {
+	Transaction types.Transaction `json:"transaction"`
+}
+
+type WalletTransactions struct {
+	Transactions []wallet.Transaction `json:"transactions"`
+}
+
+type TxpoolTransactions struct {
+	Transactions []types.Transaction `json:"transactions"`
+}
+
+type SyncerPeers struct {
+	Peers []gateway.Header `json:"peers"`
+}
```

Comment:
usually we prefer to use structs as responses, since you can add more fields to them later without breaking compatibility. But for simple list-type responses like transactions, peers, etc., I think it's fine to use a slice. In other words:
```go
type WalletAddresses []types.Address
type WalletTransactions []wallet.Transaction
type SyncerPeers []gateway.Header
// etc.
```